File: TP01/Q01/palindromo.java
import java.util.Scanner;

public class palindromo {
    public static void main(String[] args) {
        // ler entrada linha por linha e checar se e palindromo ou fim do arquivo
        Scanner sc = new Scanner(System.in);
        String str = sc.nextLine();
        while (isFim(str) == false) {
            isPalindromo(str);
            str = sc.nextLine();
        }
        sc.close();
    }

    // checa se e palindromo ou nao e imprime o resultado
    public static void isPalindromo(String str) {
        String palindromo = "SIM";
        for (int i = 0; i < str.length() / 2; i++) {
            if (str.charAt(i) != str.charAt(str.length() - 1 - i)) {
                palindromo = "NAO";
                i = str.length();
            }
        }
        System.out.println(palindromo);
    }

    // checa se o arquivo chegou ao fim
    public static boolean isFim(String str) {
        return ((str.length() == 3) && (str.charAt(0) == 'F') && (str.charAt(1) == 'I')
                && (str.charAt(2) == 'M'));
    }
}

File: TP01/Q02/palindromo.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
void isPalindromo(int tam, char str[]);
int isFim(char str[]);
int main()
{
    // VARIAVEIS
    char str[250];
    int tam = 0;

    fgets(str, 250, stdin);
    tam = strlen(str);
    while (isFim(str) == 0)
    {
        isPalindromo(tam, str);
        fflush(stdin);
        fgets(str, 250, stdin);
        tam = strlen(str);
    }
    return 1;
}
// checa se e palindromo ou nao e imprime o resultado na tela
void isPalindromo(int tam, char str[])
{
    int teste = 0;
    for (int i = 0; i < tam - 2 / 2; i++)
    {
        if (str[i] != str[tam - 2 - i])
        {
            teste = 1;
            i = tam;
        }
    }
    if (teste == 0)
    {
        printf("SIM\n");
    }
    else
    {
        printf("NAO\n");
    }
}
// checa se chegou ao fim do input
int isFim(char str[])
{
    return ((str[0] == 'F') && (str[1] == 'I') && (str[2] == 'M'));
}